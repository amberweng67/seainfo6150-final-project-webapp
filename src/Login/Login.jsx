import React, {useState} from 'react';
import styles from './Login.module.css';
import { useHistory } from "react-router-dom";

const Login = ({setUser}) => {
    const [userName, setUserName] = useState('');
    const [password, setPassword] = useState('');
    const [remember, setRemember] = useState(true);
    const history = useHistory();
    const login = () => {
        setUser(userName);
        history.push('/');
    }
    return (
        <div>

            <form className={styles.formContainer}>
            <div className={styles.container}>
            <label htmlFor="uname"><b>Username</b></label>
            <input id="uname" type="text" placeholder="Enter Username" name="uname" required value={userName} onChange={({target: {value}}) => setUserName(value)}/>
            <label htmlFor="psw"><b>Password</b></label>
            <input id="psw" type="password" placeholder="Enter Password" name="psw" required value={password} onChange={({target: {value}}) => setPassword(value)}/>
            <button onClick={login}>Login</button>
            
            <label>
                <input type="checkbox" checked={remember} name="remember" onChange={() => {setRemember(!remember)}}/> Remember me
            </label>
            </div>
            </form>
        </div>
    )
}

export default Login;
