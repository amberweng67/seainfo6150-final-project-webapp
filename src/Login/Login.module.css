/* Bordered form */
form {
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  transition: 0.3s;
  border-radius: 10px;
  width: 50%;
  margin: auto;
}

input[type=text], input[type=password] {
  width: 100%;
  padding: 16px 20px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  box-sizing: border-box;
}

button {
  background-color: #001a3b;
  color: white;
  padding: 14px 20px;
  margin: 8px 0;
  border: none;
  cursor: pointer;
  width: 100%;
  border-radius: 20px;
}

button:hover {
  opacity: 0.8;
}

.container {
  padding: 16px;
  margin: 16px;
}

@media screen and (max-width: 600px) {
  .formContainer  {
    width: 95%;
  }
}

@media screen and (max-width: 960px) and (min-width:601px) {
  .formContainer  {
    width: 75%;
  }
}
