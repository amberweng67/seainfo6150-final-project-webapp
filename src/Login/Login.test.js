import React from "react";
import Login from "./Login.jsx";

it("renders the login component correctly", () => {
  const { container } = render(<Login />);
  expect(container).toMatchSnapshot();
});
