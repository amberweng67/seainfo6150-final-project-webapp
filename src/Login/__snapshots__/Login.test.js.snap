// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the login component correctly 1`] = `
<div>
  <div>
    <form>
      <div
        class="container"
      >
        <label
          for="uname"
        >
          <b>
            Username
          </b>
        </label>
        <input
          name="uname"
          placeholder="Enter Username"
          required=""
          type="text"
          value=""
        />
        <label
          for="psw"
        >
          <b>
            Password
          </b>
        </label>
        <input
          name="psw"
          placeholder="Enter Password"
          required=""
          type="password"
          value=""
        />
        <button>
          Login
        </button>
        <label>
          <input
            checked=""
            name="remember"
            type="checkbox"
          />
           Remember me
        </label>
      </div>
    </form>
  </div>
</div>
`;
